src/auth/mod.ts

```diff
@@ -13,35 +15,169 @@ interface DecodedToken {
   };
 }
 
+interface JWK {
```

Comment:
this is the structure of the "kube" jwt specifically, include that in the name of the type? and comment the full name of every field?